apps/enterprise-knowledge-retrieval/enterprise_knowledge_retrieval.ipynb

```diff
@@ -0,0 +1,1750 @@
+    "    evaluation_output.append((pair[0],pair[1],evaluation['choices'][0]['message']['content']))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 97,
+   "id": "f18678e0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def collate_results(x):\n",
+    "    text = x.lower()\n",
+    "    \n",
+    "    if 'incorrect' in text:\n",
+    "        return 'incorrect'\n",
+    "    elif 'correct' in text:\n",
+    "        return 'correct'\n",
+    "    else: \n",
+    "        return 'unable to answer'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 98,
+   "id": "cd29fc90",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "correct             50\n",
+       "unable to answer    28\n",
+       "incorrect           5 \n",
+       "Name: evaluation, dtype: int64"
+      ]
+     },
+     "execution_count": 98,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "eval_df = pd.DataFrame(evaluation_output)\n",
+    "eval_df.columns = ['question','answer','evaluation']\n",
+    "# Replacing all the \"unable to evaluates\" with \"unable to answer\"\n",
+    "eval_df['evaluation'] = eval_df['evaluation'].apply(lambda x: collate_results(x))\n",
+    "eval_df.evaluation.value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "566053d6",
+   "metadata": {},
+   "source": [
+    "#### Analysis\n",
+    "\n",
+    "We ended up with a 60% first hit rate, which isn't great, but we've got a baseline to work from.\n",
+    "\n",
+    "Our remediation plan could be as follows:\n",
+    "- **Incorrect answers:** Either prompt engineering to help the model work out how to answer better (maybe even a bigger model like GPT-4), or search optimisation to return more relevant chunks. Chunking/embedding changes may help this as well - larger chunks may give more context, allowing the model to formulate a better answer.\n",
+    "- **Unable to answer:** This is either a retrieval problem, or the data doesn't exist in our knowledge base. We can prompt engineer to classify questions that are \"out-of-bounds\", or we can tune our search so the relevant data is returned.\n",
+    "\n",
+    "This is the framework we'll build on to get our knowledge retrieval solution to production - again, log everything and store each run down to a question level so you can track regressions and iterate towards your production solution."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5016fcb8",
+   "metadata": {},
+   "source": [
+    "## Conclusion\n",
+    "\n",
+    "This concludes our Enterprise Knowledge Retrieval walkthrough. We hope you've found it useful, and that you're now in a position to scale your knowledge retrieval solutions into production confidently.\n",
+    "\n",
+    "Let us know what you think, and if you have any questions that you'd like answered then please sign up for the Q&A webinar [here](link TBD)."
```

Comment:
If we're asking for feedback, we should tell folks where and how to give us feedback. Issues page? Discussion forum?

Reply:
I've removed this now